### PR TITLE
Fix/optional trailing dot kube

### PR DESCRIPTION
**Pull request by szuecs**
Kubernetes generated routes have optional port but not optional trailing dot, such that

`example.org.` in the host header would not match.
There is a longer discussion in https://lists.w3.org/Archives/Public/ietf-http-wg/2016JanMar/0447.html that turns out to show that proxies and some clients remove trailing dot in SNI and Host header which is good, but we don't fix bogus clients in this case.

A follow up is to create a filter that removes the trailing dot from the host header, such that in case preserHost(true) we would send the host header without trailing dot if the client sends a trailing dot

### CODE REVIEW

File: dataclients/kubernetes/hosts.go [hunk 1]
```diff
@@ -15,7 +15,10 @@ func createHostRx(hosts ...string) string {
 
 	hrx := make([]string, len(hosts))
 	for i, host := range hosts {
-		hrx[i] = strings.Replace(host, ".", "[.]", -1) + "(:[0-9]+)?"
+		// trailing dots and port are not allowed in kube
+		// ingress spec, so we can append optional setting
+		// without check
+		hrx[i] = strings.Replace(host, ".", "[.]", -1) + "([.])?(:[0-9]+)?"
```
Comment (AlexanderYastrebov) [inline dataclients/kubernetes/hosts.go:21]: I think there is no need to have a capture group around the dot, i.e. `[.]?` should work

Comment (szuecs) [reply dataclients/kubernetes/hosts.go]: fixed

**Comment by AlexanderYastrebov** [comment]
:+1:

**Comment by szuecs** [comment]
:+1:

**Pull request merged**